index.bs

```diff
@@ -795,6 +743,562 @@ the following list:
+    requirements for elements that comprise the value of
+    \`<a http-header>`Sec-WebSocket-Protocol`</a>\` fields as defined by The WebSocket protocol,
+    then throw a "{{SyntaxError}}" {{DOMException}}.  [[!WSP]]
+ 1. Set [=this=]'s [=WebSocketStream/url=] to |urlRecord|.
+ 1. Set [=this=]'s [=WebSocketStream/opened promise=] and [=WebSocketStream/closed promise=] to new
+    promises.
+ 1. Apply backpressure to the WebSocket.
+
+  Note: This means that messages won't be read until the application is ready for them.
+
+ 1. If |options|["{{WebSocketStreamOptions/signal}}"] [=map/exists=],
+  1. Let |signal| be |options|["{{WebSocketStreamOptions/signal}}"].
+  1. If |signal| is [=AbortSignal/aborted=], then reject [=this=]'s [=WebSocketStream/opened
+     promise=] and [=WebSocketStream/closed promise=] with |signal|'s [=abort reason=] and return.
+  1. [=AbortSignal/add|Add the following abort steps=] to |signal|:
+   1. If the WebSocket connection is not yet [=established=] [[!WSP]]
+    1. [=Fail the WebSocket connection=] and set [=this=]'s [=WebSocketStream/ready state=] to
+       {{WebSocket/CLOSING}}.
+    1. [=Reject=] [=this=]'s [=WebSocketStream/opened promise=] and [=WebSocketStream/closed
+       promise=] with |signal|'s [=abort reason=].
+    1. Set [=this=]'s [=WebSocketStream/handshake aborted flag=].
+ 1. Let |client| be [=this=]'s [=relevant settings object=].
+ 1. Run this step [=in parallel=]:
+  1. [=Establish a WebSocket connection=] given |urlRecord|, |protocols|, and |client|. [[!FETCH]]
+
+     <p class="note">If the [=establish a WebSocket connection=] algorithm fails, it triggers the
+     [=fail the WebSocket connection=] algorithm, which then invokes the [=close the WebSocket
+     connection=] algorithm. This establishes that [=the WebSocket connection is closed=], which
+     <a>rejects</a> the {{WebSocketStream/opened}} and {{WebSocketStream/closed}} promises.
+</div>
+
+<hr>
+
+The <dfn attribute for=WebSocketStream>url</dfn> getter steps are to return [=this=]'s
+[=WebSocketStream/url=], [=URL serializer|serialized=].
+
+The <dfn attribute for=WebSocketStream>opened</dfn> getter steps are to return [=this=]'s
+[=WebSocketStream/opened promise=].
+
+The <dfn attribute for=WebSocketStream>closed</dfn> getter steps are to return [=this=]'s
+[=WebSocketStream/closed promise=].
+
+<div algorithm>
+ The <dfn method for=WebSocketStream>close(|closeInfo|)</dfn> method steps are:
+
+ 1. Let |code| be |closeInfo|["{{WebSocketCloseInfo/closeCode}}"] if present, or null otherwise.
+ 1. Let |reason| be |closeInfo|["{{WebSocketCloseInfo/reason}}"].
+ 1. [=Close the WebSocket=] with <var ignore>this</var>, |code|, and |reason|.
+</div>
+
+
+## Feedback to WebSocketStream from the protocol ## {#feedback-to-websocket-stream-from-the-protocol}
+
+When [=the WebSocket connection is established=] for a {{WebSocketStream}} |stream|, the user agent
+must [=queue a task=] to run these steps:
```

Comment:
Can you be more specific about the task-queuing, using https://html.spec.whatwg.org/#queue-a-global-task and in particular picking a task source? (Here and below.)

Reply:
Done. I copied the wording from the existing WebSocket API algorithms, so those should probably be updated in a separate PR.